source/AI.cpp

```diff
@@ -322,11 +322,16 @@ void AI::Step(const PlayerInfo &player)
 			++it;
 	}
 	for(const auto &it : ships)
-		if(it->Position().Length() >= MAX_DISTANCE_FROM_CENTER)
+    {
+	    double border = 0;
+	    if(it->GetSystem())
+            border = it->GetSystem()->Border();
+		if(it->Position().Length() >= MAX_DISTANCE_FROM_CENTER+border)
```

Comment:
You need spaces between binary operators like `+`

Additionally, I don't think it is correct to add these values. You likely want to consider only the maximum value.